tools/small_rna_map/small_rna_map.py

```diff
@@ -35,28 +35,28 @@ def Ref_coord_pol(bamfile):
                         read_rname = read.reference_name
                         read_position = read.pos + 1
                         the_key = (read_rname, read_position,polarity)
+                        cigar = read.cigar[0][1]
```

Comment:
I think this method .cigar is tagged as deprecated (http://pysam.readthedocs.io/en/latest/api.html)
In addition are you sure that is always return the read **length** ?
Why not using the pysam method query_length in the class pysam.AlignedSegment ? -- pysam.AlignedSegment.query_length